chains/solana/utils/common/pda.go

```diff
@@ -0,0 +1,71 @@
+package common
+
+import (
+	"encoding/binary"
+
+	"github.com/gagliardetto/solana-go"
+)
+
+func FindConfigPDA(programID solana.PublicKey) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("config")}, programID)
+}
+
+func FindStatePDA(programID solana.PublicKey) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("state")}, programID)
+}
+
+func FindExternalExecutionConfigPDA(programID solana.PublicKey) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("external_execution_config")}, programID)
+}
+
+func FindExternalTokenPoolsSignerPDA(programID solana.PublicKey) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("external_token_pools_signer")}, programID)
+}
+
+func FindCounterPDA(programID solana.PublicKey) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("counter")}, programID)
+}
+
+func FindFeeBillingSignerPDA(programID solana.PublicKey) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("fee_billing_signer")}, programID)
+}
+
+func FindTokenAdminRegistryPDA(programID, mint solana.PublicKey) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("token_admin_registry"), mint.Bytes()}, programID)
+}
+
+func FindFeeBillingTokenConfigPDA(programID, mint solana.PublicKey) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("fee_billing_token_config"), mint.Bytes()}, programID)
+}
+
+func FindCcipTokenpoolBillingPDA(programID, mint solana.PublicKey, chainSelector uint64) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("ccip_tokenpool_billing"), binary.LittleEndian.AppendUint64([]byte{}, chainSelector), mint.Bytes()}, programID)
+}
+
+func FindCcipTokenpoolChainconfigPDA(programID, mint solana.PublicKey, chainSelector uint64) (solana.PublicKey, uint8, error) {
+	return solana.FindProgramAddress([][]byte{[]byte("ccip_tokenpool_chainconfig"), binary.LittleEndian.AppendUint64([]byte{}, chainSelector), mint.Bytes()}, programID)
+}
+
+func GetSourceChainStatePDA(chainSelector uint64, ccipRouterProgram solana.PublicKey) (solana.PublicKey, error) {
```

Comment:
Please modify all methods to respect the same format, the params of them should be in the same order as they are serialized in the seeds, and also they all should be `ccipRouterProgram` not just `programID`